### PR TITLE
Minor update to dataset docs to lead users to quickstart and list_datasets

### PR DESCRIPTION
**Pull request by DSGuala**
**Context:**
Users have trouble finding all `bondlength` and `basis` combinations for qchem molecules. They end up using the slider on the website to check which bond lengths are available one-by-one.
image

**Description of the Change:**
- Updated the quickstart guide to say "viewing available datasets" instead of "dataset structure" 
- Updated the `qml.data` docs to encourage users to first read the quickstart guide 

**Benefits:**
Users should be more likely to reach the `list_datasets` description in the quickstart guide and know that it can be used to view the available datasets.

**Possible Drawbacks:**
Users who use pennylane.ai/datasets but don't go to the documentation still won't see this.

### CODE REVIEW

**Comment by trbromley** [comment]
@DSGuala is this needed for the v0.36 release?

**Comment by DSGuala** [comment]
> @DSGuala is this needed for the v0.36 release?

@trbromley  Not required. We can leave it for the v0.37 release.

**Comment by josh146** [comment]
@DSGuala just to clarify, is this PR complete or did you still have changes you want to make? Because if it is complete, we could merge during feature freeze next week

**Comment by DSGuala** [comment]
> @DSGuala just to clarify, is this PR complete or did you still have changes you want to make? Because if it is complete, we could merge during feature freeze next week

@josh146 this is basically completed. But I haven't had the bandwidth to finalize it or get it reviewed.

**Comment by DSGuala** [comment]
> @DSGuala would it make sense to come back to this now and get it merged in soon?

Yup! I will take a look today.

**Review by trbromley — APPROVED** [review]
Approved this PR @DSGuala! Though let me know if you'd like to change anything and I'll take another look.

File: doc/introduction/data.rst [hunk 1]
```diff
@@ -152,6 +152,9 @@ We can then write this :class:`~pennylane.data.Dataset` to storage and read it a
 >>> read_dataset.energies
 array([-1.5, -0.5,  0.5,  1.5])
 
+For more details on reading and writing custom datasets, including metadata, please
+see the :mod:`~.data` subpackage documentation.
```
Comment (obliviateandsurrender) [inline doc/introduction/data.rst:154]:
```suggestion
see the :mod:`~.data` module documentation.
```

**Review by obliviateandsurrender — APPROVED** [review]
Looks good! Just one optional suggestion.

> Users who use pennylane.ai/datasets but don't go to the documentation still won't see this.

Regarding the drawback, a couple of suggestions:
1. We can link the quick start page in the download dialogue box as hyperlinked via  `?` or `More details`. Otherwise, you may add a code snippet to know more about that particular dataset using `list_dataset`. (the second option might be uglier)
2. We should arrange a centralized tabulated- or file-explorer-like page to showcase a snapshot of all datasets we host based on the website. You may think of it as a prettier version of `list_datasets`, which we can simply link to.

**Pull request merged**